packages/twenty-server/src/engine/core-modules/billing-webhook/services/billing-webhook-subscription-schedule.service.ts

```diff
@@ -0,0 +1,63 @@
+/* @license Enterprise */
+
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+import Stripe from 'stripe';
+import { isDefined } from 'twenty-shared/utils';
+
+import { BillingSubscription } from 'src/engine/core-modules/billing/entities/billing-subscription.entity';
+import { StripeSubscriptionScheduleService } from 'src/engine/core-modules/billing/stripe/services/stripe-subscription-schedule.service';
+import { transformStripeSubscriptionScheduleEventToDatabaseSubscriptionPhase } from 'src/engine/core-modules/billing-webhook/utils/transform-stripe-subscription-schedule-event-to-database-subscription-phase.util';
+
+@Injectable()
+export class BillingWebhookSubscriptionScheduleService {
+  protected readonly logger = new Logger(
+    BillingWebhookSubscriptionScheduleService.name,
+  );
+
+  constructor(
+    @InjectRepository(BillingSubscription)
+    private readonly billingSubscriptionRepository: Repository<BillingSubscription>,
+    private readonly stripeSubscriptionScheduleService: StripeSubscriptionScheduleService,
+  ) {}
+
+  async processStripeEvent(
+    data:
+      | Stripe.SubscriptionScheduleUpdatedEvent.Data
+      | Stripe.SubscriptionScheduleCanceledEvent.Data,
+  ) {
+    const schedule = data.object as Stripe.SubscriptionSchedule;
+
+    if (!isDefined(schedule.subscription)) {
+      throw new Error('Subscription is not defined');
+    }
+
+    const subscriptionId =
+      typeof schedule.subscription === 'string'
```

Comment:
let's avoid this! schedule.subscription can actually be an id? is stripe API inconsistent?